public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx

```diff
@@ -151,16 +157,12 @@ export const AlertRuleForm = ({ existing, prefill }: Props) => {
     );
   };
 
-  const deleteRule = () => {
+  const deleteRule = async () => {
     if (existing) {
-      const identifier = ruleId.fromRulerRule(
-        existing.ruleSourceName,
-        existing.namespace,
-        existing.group.name,
-        existing.rule
-      );
-
-      dispatch(deleteRuleAction(identifier, { navigateTo: '/alerting/list' }));
+      const ruleGroupIdentifier = getRuleGroupLocationFromRuleWithLocation(existing);
+
+      await deleteRuleFromGroup(ruleGroupIdentifier, existing.rule);
```

Comment:
(Maybe a `try/catch` here as well?)

Reply:
I think in this particular case we are not throwing any error. Not sure if we should wrap it with try/catch though 🤔

Reply:
This one does throw if it fails – but the `ConfirmModal` handles Promises for us :)

https://github.com/grafana/grafana/blob/52a824f4ed6a23f74ddaf89ab13f74b8ff308799/packages/grafana-ui/src/components/ConfirmModal/ConfirmContent.tsx#L77-L81